Add set_env option

uzxmx opened the pull request:
Closes #725.

This PR adds a new SSH option `set_env` that is supported in [OpenSSH 7.8](https://www.openssh.com/txt/release-7.8). For more info, please see #725.

mfazekas commented on lib/net/ssh/connection/channel.rb:
```diff
@@ -677,6 +678,16 @@ def forward_local_env(env_variable_patterns)
+        #
+        #   channel.set_remote_env foo: 'bar', baz: 'whale'
+        #
+        # rubocop:disable Naming/AccessorMethodName
```
pls just disable Naming/AccessorMethodName from config

uzxmx replied:
Got it. Meanwhile, I'm trying to add some integration test for what I've done, and upgrade OpenSSH used in integration test from 7.4p1 to 7.9p1. Because I saw the codecov/patch check was not passed, seemed the test coverage was not enough. Will let you have a look after I finish.

mfazekas replied:
It's complining that `set_remote_env` in channel.rb is not tested you might get away without integation test but testing that line, but sure an integration would be great!

uzxmx commented on test/integration/common.rb:
```diff
@@ -94,6 +94,7 @@ def with_lines_as_tempfile(lines = [], &block)
 
   # @yield [pid, port]
   def start_sshd_7_or_later(port = '2200', config: nil)
+    pid = nil
```
Without pre-declaring `pid`, when `config` exists, the `pid` within `with_lines_as_tempfile` block will be out of scope, which results that statements in `ensure` part doesn't execute as expected (in other words, sshd process won't be killed).

mfazekas replied:
good catch thanks!

uzxmx commented on test/integration/test_channel.rb:
```diff
@@ -51,11 +51,11 @@ def test_transport_close_before_channel_close_should_raise
       res = nil
       Net::SSH.start(*ssh_start_params(proxy: proxy)) do |ssh|
-        chanell_success_handler = lambda do
+        channel_success_handler = lambda do
```
Fix typo

uzxmx commented on test/integration/test_forward.rb:
```diff
@@ -24,18 +24,6 @@
   include IntegrationTestHelpers
 
-  # @yield [pid, port]
-  def start_sshd_7_or_later(port = '2200')
```
Remove this method because it's already been implemented in `IntegrationTestHelpers` that is mixed in.

uzxmx commented:
@mfazekas I've disabled rubocop offenses from config. Also I've added an integration test case for new `set_env` option. When I implemented that, I found a trivial bug and some typos, also removed a duplicated method definition. Detailed information is added as inline comment.
